Development comparisons

jhadobe opened the pull request:
This PR adds comparison info between in-process and out-of-process extensibility. Highlighting what is available in Commerce vs what is achievable with App Builder + Commerce.

staging:
- https://adobedocs.github.io/commerce-extensibility/app-development/
- https://adobedocs.github.io/commerce-extensibility/app-development/app-development-comparison/
- https://adobedocs.github.io/commerce-extensibility/app-development/services-comparison/

jhadobe commented:
A few questions for @keharper:

- Do you have ideas for better page names?
- Should I break the third-party integrations section into a separate page?
- `app-development-comparison` looks a little flat due to the similarity between H2/H3, but the table structure was not conducive either. Any ideas on visually separating these sections? All I can think of is to use more unordered lists, like in `services-comparison`.md.
- How do we want to credit the developer?

keharper commented on src/pages/_images/oms2.mmd:
```diff
@@ -0,0 +1,10 @@
+flowchart LR
+    subgraph Box1[Adobe Commerce]
+        A1[New Order] --> A2[I/O Events to Adobe Commerce]
+        A3[AC Extenstion that adds custom data to GQL or custom calcultations] --> A4[GQL]
```
GH isn't letting me make a suggestion here.

Change line 4 to 

A3[Extension that adds custom data to GraphQL or performs custom calculations] --> A4[GraphQL]

nuzil commented on src/pages/app-development/port-extensions.md:
```diff
@@ -0,0 +1,75 @@
+*  Database data
+*  Filesystem
+
+### Frontend development
```
how about to add here a note that native HTML + CSS is also an option?

nuzil commented:
Will pass it also to @keharper 
We discussed that we need to add 2 sections for 

1. Instead of treating in-process and out-of-process are equals, we should be clear that unless absolutely needed, we recommend developers always choose out of process
2. In services comparison, we should be clear that modifying the default MySQL database can cause upgrade issues, so we recommend using App Builder with state database for simple app specific data, and a 3rd party database for more complex needs.

My suggestion is to add block for point 1: 

```
In-process extensibility model currently has a lot of disaadvantages like, increasing maintenance costs, reducing software stability, increasing AC server overload and more.
Looking on those factors and also on increasign of business compexity and technologies stack, we recommends to use out-of-process model as much as it possible
```

And similar for 2nd

jhadobe commented:
> Will pass it also to @keharper We discussed that we need to add 2 sections for
> 
> 1. Instead of treating in-process and out-of-process are equals, we should be clear that unless absolutely needed, we recommend developers always choose out of process
> 2. In services comparison, we should be clear that modifying the default MySQL database can cause upgrade issues, so we recommend using App Builder with state database for simple app specific data, and a 3rd party database for more complex needs.
> 
> My suggestion is to add block for point 1:
> 
> ```
> In-process extensibility model currently has a lot of disaadvantages like, increasing maintenance costs, reducing software stability, increasing AC server overload and more.
> Looking on those factors and also on increasign of business compexity and technologies stack, we recommends to use out-of-process model as much as it possible
> ```
> 
> And similar for 2nd

Thanks @nuzil I had already added bolded "Recommended method" text to all the "App Builder" sections to help clearly emphasize that OOP is the best option. I've also added a version of your quote above.

slamech commented on src/pages/app-development/index.md:
"Each approach has its own set of advantages and use cases, and understanding them is crucial for developers and businesses looking to maximize their platform's potential" - Can we change this line to say something like "Adobe recommends adopting out-of-process extensibility where possible due to its ease of maintenance."

slamech commented on src/pages/app-development/index.md:
```diff
@@ -1,75 +1,134 @@
+# Extensibility in Adobe Commerce
 
-Developers have traditionally used PHP to create in-process extensions that add features, functionality, services, and integrations to Adobe Commerce. This model requires any new code to be compatible with upgrades, the server's PHP version, and many other essential server applications and services that Commerce uses.
+Adobe Commerce is known for its flexibility and extensibility, allowing businesses to adapt their e-commerce platforms to their unique needs. With the evolving demands of e-commerce, Adobe Commerce offers various methods for extending its capabilities. These methods can be broadly categorized into **in-process, out-of-process, and hybrid** extensibility. Each approach has its own set of advantage ...
```
```suggestion
Adobe Commerce is known for its flexibility and extensibility, allowing businesses to adapt their e-commerce platforms to their unique needs. With the evolving demands of e-commerce, Adobe Commerce offers various methods for extending its capabilities. These methods can be broadly categorized into **in-process, out-of-process, and hybrid** extensibility. Adobe recommends using the out-of-process extensibility method where possible due to its ease of maintenance.
```

slamech commented on src/pages/app-development/index.md:
```diff
@@ -1,75 +1,134 @@
+* **Technology agnostic** - Developers can use different programming languages or frameworks for their extensions, providing greater flexibility.
 
-App Builder uses Apache OpenWhisk Alarms to perform the scheduling services traditionally provided by cron jobs. [Scheduling Cron Jobs with Alarms](https://developer.adobe.com/app-builder/docs/resources/cron-jobs/) walks you through the process of implementing this feature.
+However, setting up and managing out-of-process extensions can be more complex, requiring additional infrastructure and communication protocols. For example, you might have additional database requirements, Elasticsearch set up, etc.
```
Not sure why we have this line here since we provide the infrastructure (App Builder / API Mesh) to host the out of process extensions, Can we remove this and instead add a line something like "When choosing this out of process extensibility, Adobe provides developers with serverless runtime to host their apps reducing the overhead for developers"

slamech commented on src/pages/app-development/app-development-comparison.md:
```diff
@@ -0,0 +1,179 @@
+
+# App Development Comparison
+
+Adobe Commerce is a powerful e-commerce platform that allows developers to use in-process extensibility to enhance its functionality. By combining Commerce with Adobe Developer App Builder, developers now have a new out-of-process set of tools. This guide highlights key differences and benefits between using Commerce's built-in extensibility offerings and App Builder's [out-of-process development ...
```
```suggestion
Adobe Commerce is a powerful e-commerce platform that allows developers to use in-process extensibility to enhance its functionality. By combining Commerce with Adobe Developer App Builder, developers now have a new out-of-process set of tools. This guide highlights key differences and benefits of using Commerce's out-of-process extensibility with App Builder's [out-of-process development tools](../index.md).
```

slamech commented on src/pages/app-development/services-comparison.md:
```diff
@@ -0,0 +1,169 @@
+
+# Services comparison
+
+This guide highlights key differences and benefits between Adobe Commerce and Adobe Developer App Builder services. It provides a comparison of services used in both platforms and outlines the differences in their implementation and usage.
```
```suggestion
This guide highlights key differences and benefits of using Adobe Developer App Builder services over services available within Adobe Commerce. It provides a comparison of services used in both platforms and outlines the differences in their implementation and usage.
```
